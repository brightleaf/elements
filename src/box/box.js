import React from 'react'
import classnames from 'classnames'
import { Base } from '../base'
import { propTypes } from '../utils/prop-types'

const Box = ({ children, className, ...props }) => {
  return (
    <Base className={classnames('box', className)} {...props}>
      {children}
    </Base>
  )
}

Box.propTypes = propTypes

Box.defaultProps = {
  as: 'div',
}

export { Box }

export default Box
