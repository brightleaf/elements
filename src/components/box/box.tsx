import React, { FC } from "react";
import classnames from "classnames";
import { Base, BaseProps } from "../base";

export const Box: FC<BaseProps> = ({
  as = "div",
  className,
  children,
  ...props
}) => {
  return (
    <Base as={as} className={classnames("box", className)} {...props}>
      {children}
    </Base>
  );
};

Box.defaultProps = {
  as: "div",
};

export default Box;
