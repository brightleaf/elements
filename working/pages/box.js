/* eslint-disable spellcheck/spell-checker */
/* eslint-disable sonarjs/no-duplicate-string */
import React from 'react'
import { useStyleSheet } from '@brightleaf/react-hooks/lib/use-stylesheet'
import { useStyles } from '@brightleaf/react-hooks/lib/use-styles'
import Highlight from '../components/highlighter'
import {
  Box,
  Content,
  Columns,
  Column,
  Delete,
  Hero,
  HeroBody,
  Icon,
  Image,
  Level,
  LevelItem,
  LevelLeft,
  MediaContent,
  MediaLeft,
  MediaObject,
  MediaRight,
  Section,
  SubTitle,
  Title,
} from '../../src'
import { Snippet } from '../components/snippet'
import { ElementsTabs } from '../components/tabs'

export default () => {
  useStyleSheet('https://fonts.googleapis.com/css?family=Open+Sans')
  useStyles(`
  html,
  body {
      font-family: 'Open Sans';
  }
  `)
  useStyleSheet('code.css')
  return (
    <Section>
      <Hero isLight isBold>
        <HeroBody>
          <Title>Box</Title>
          <SubTitle as="p" is="4">
            A white <strong>box</strong> to contain other elements
          </SubTitle>
        </HeroBody>
      </Hero>

      <ElementsTabs />
      <Snippet>
        <Columns>
          <Column isHalf className="snippet-preview">
            <Box>
              <MediaObject>
                <MediaLeft>
                  <Image
                    is="64"
                    src="https://bulma.io/images/placeholders/128x128.png"
                  />
                </MediaLeft>
                <MediaContent>
                  <Content>
                    <p>
                      <strong>John Smith</strong> <small>@johnsmith</small>{' '}
                      <small>31m</small>
                      <br />
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                      Proin ornare magna eros, eu pellentesque tortor vestibulum
                      ut. Maecenas non massa sem. Etiam finibus odio quis
                      feugiat facilisis.
                    </p>
                  </Content>
                  <Level isMobile>
                    <LevelLeft>
                      <LevelItem>
                        <a>
                          <Icon isSmall fas icon="reply"></Icon>
                        </a>
                      </LevelItem>
                      <LevelItem>
                        <a>
                          <Icon isSmall fas icon="retweet"></Icon>
                        </a>
                      </LevelItem>
                      <LevelItem>
                        <a>
                          <Icon isSmall fas icon="heart"></Icon>
                        </a>
                      </LevelItem>
                    </LevelLeft>
                  </Level>
                </MediaContent>
                <MediaRight>
                  <Delete></Delete>
                </MediaRight>
              </MediaObject>
            </Box>
          </Column>
          <Column isHalf>
            <Highlight className="javascript" languages={['javascript']}>
              {`
import React from 'react'
import {
  Box, Content, Columns, Column, Delete, Icon, Image, Level, LevelItem,
  LevelLeft, MediaContent, MediaLeft, MediaObject, MediaRight
} from '@brightleaf/elements'


export default () => {

  return (
    <Box>
      <MediaObject>
        <MediaLeft>
          <Image
            is="64"
            src="https://bulma.io/images/placeholders/128x128.png"
          />
        </MediaLeft>
        <MediaContent>
          <Content>
            <p>
              <strong>John Smith</strong> <small>@johnsmith</small>{' '}
              <small>31m</small>
              <br />
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin
              ornare magna eros, eu pellentesque tortor vestibulum ut.
              Maecenas non massa sem. Etiam finibus odio quis feugiat
              facilisis.
            </p>
          </Content>
          <Level isMobile>
            <LevelLeft>
              <LevelItem>
                <a>
                  <Icon isSmall fas icon="reply"></Icon>
                </a>
              </LevelItem>
              <LevelItem>
                <a>
                  <Icon isSmall fas icon="retweet"></Icon>
                </a>
              </LevelItem>
              <LevelItem>
                <a>
                  <Icon isSmall fas icon="heart"></Icon>
                </a>
              </LevelItem>
            </LevelLeft>
          </Level>
        </MediaContent>
        <MediaRight>
          <Delete></Delete>
        </MediaRight>
      </MediaObject>
    </Box>
  )
}
`}
            </Highlight>
          </Column>
        </Columns>
      </Snippet>
    </Section>
  )
}
