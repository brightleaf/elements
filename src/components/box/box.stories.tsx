import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Box } from './box'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Elements/Box',
  component: Box,
} as ComponentMeta<typeof Box>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Box> = (args) => <Box {...args} />

export const Primary = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = {
  children: 'Box Text',
}

export const AsASpan = Template.bind({})
AsASpan.args = {
  children: 'Box Text',
  as: 'span',
}
