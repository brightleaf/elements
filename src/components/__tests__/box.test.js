import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { toHaveClass } from '@testing-library/jest-dom'
import { Box } from '../box'

expect.extend({ toHaveClass })

describe('Box component', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(
      <Box>
        <div>Trapped in a Box</div>
      </Box>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('box')
  })
  it('should render with extra classnames', () => {
    const { container } = render(
      <Box className="boxed-in">
        <div>Trapped in a Box</div>
      </Box>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('box')
    expect(container.firstChild).toHaveClass('boxed-in')
  })
  it('should render with has-background-primary when hasBackgroundPrimary set', () => {
    const { container } = render(
      <Box className="boxed-in" hasBackgroundPrimary>
        <div>Trapped in a Box</div>
      </Box>
    )

    expect(container.firstChild).toHaveClass('box')
    expect(container.firstChild).toHaveClass('has-background-primary')
  })
  test.each([
    ['has-background-primary', 'hasBackgroundPrimary'],
    ['has-background-success', 'hasBackgroundSuccess'],
    ['has-background-warning', 'hasBackgroundWarning'],
    ['has-background-danger', 'hasBackgroundDanger'],
    ['has-background-info', 'hasBackgroundInfo'],
    ['has-background-white', 'hasBackgroundWhite'],
    ['has-background-black', 'hasBackgroundBlack'],
    ['has-background-light', 'hasBackgroundLight'],
    ['has-background-dark', 'hasBackgroundDark'],
    ['has-background-link', 'hasBackgroundLink'],
    ['has-background-black-bis', 'hasBackgroundBlackBis'],
    ['has-background-black-ter', 'hasBackgroundBlackTer'],
    ['has-background-grey-darker', 'hasBackgroundGreyDarker'],
    ['has-background-grey-dark', 'hasBackgroundGreyDark'],
    ['has-background-grey', 'hasBackgroundGrey'],
    ['has-background-grey-light', 'hasBackgroundGreyLight'],
    ['has-background-grey-lighter', 'hasBackgroundGreyLighter'],
    ['has-background-white-ter', 'hasBackgroundWhiteTer'],
    ['has-background-white-bis', 'hasBackgroundWhiteBis'],
  ])(
    'The background color class %i is added when %i is set)',
    (className, propertyName) => {
      const prop = {}
      prop[propertyName] = true
      const { container } = render(
        <Box className="boxed-in" {...prop}>
          <div>Trapped in a Box</div>
        </Box>
      )

      expect(container.firstChild).toHaveClass('box')
      expect(container.firstChild).toHaveClass('boxed-in')
      expect(container.firstChild).toHaveClass(className)
    }
  )
})
