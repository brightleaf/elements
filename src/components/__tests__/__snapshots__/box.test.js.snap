// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component should render 1`] = `
<div
  class="box"
>
  <div>
    Trapped in a Box
  </div>
</div>
`;

exports[`Box component should render with extra classnames 1`] = `
<div
  class="box boxed-in"
>
  <div>
    Trapped in a Box
  </div>
</div>
`;
